tests/pthread/main_thread_join.cpp

```diff
@@ -0,0 +1,64 @@
+// Copyright 2019 The Emscripten Authors.  All rights reserved.
+// Emscripten is available under two separate licenses, the MIT license and the
+// University of Illinois/NCSA Open Source License.  Both these licenses can be
+// found in the LICENSE file.
+
+#include <assert.h>
+#include <emscripten.h>
+#include <pthread.h>
+#include <stdio.h>
+
+
+void *ThreadMain(void *arg)
+{
+	pthread_exit((void*)0);
+}
+
+pthread_t CreateThread()
+{
+  pthread_attr_t attr;
+  pthread_attr_init(&attr);
+  pthread_attr_setdetachstate(&attr, PTHREAD_CREATE_JOINABLE);
+  pthread_t ret;
+  int rc = pthread_create(&ret, &attr, ThreadMain, (void*)0);
+  assert(rc == 0);
+  pthread_attr_destroy(&attr);
```

Comment:
I think you can just pass NULL as attr.  Pthreads are joinable by default I pretty sure.